Create a new HttpClient per connection

4drian3d opened the pull request:
This would solve a rare ["too many concurrent streams"](https://discord.com/channels/289587909051416579/908507886420910101/1203719351136550962) error.

4drian3d committed:
Cache HttpClients instead of having a single instance

astei commented on proxy/src/main/java/com/velocitypowered/proxy/network/ConnectionManager.java:
```diff
@@ -79,9 +82,10 @@ public ConnectionManager(VelocityServer server) {
-    this.httpClient = HttpClient.newBuilder()
+    // TODO: Replace with a try-with-resources usage in Java 21
+    this.httpClientSupplier = Suppliers.memoizeWithExpiration(() -> HttpClient.newBuilder()
             .executor(this.workerGroup)
```
can we please remove this - I suspect this is causing #1242

4drian3d committed:
Create a new HttpClient for each connection

If the instance is using Java 21, the HttpClient resources will be cleaned using its AutoCloseable interface

kashike commented on proxy/src/main/java/com/velocitypowered/proxy/network/ConnectionManager.java:
```diff
@@ -238,8 +234,11 @@ public ServerChannelInitializerHolder getServerChannelInitializer() {
   }
 
+  @SuppressWarnings("checkstyle:MissingJavadocMethod")
   public HttpClient getHttpClient() {
```
```suggestion
  public HttpClient createHttpClient() {
```

4drian3d committed:
Applied suggestion
